src/main.py

```diff
@@ -30,21 +31,24 @@ def get_success_rate():
     test_lists = load_image_vectors.load_gz('../data/mnist_test.csv.gz')
     print("Successfully loaded test list")
 
+    image_operations.draw(test_lists[1].image)
+    image_operations.save('../data/fname.png', test_lists[1].image)
+    
     # Get reduces training and test images as tuple - reduced_images[0] is train_list, [1] is test_list without digits
-    reduced_images = pca.reduce_dimensions(training_lists[1], test_lists[1])
+    reduced_images = pca.reduce_dimensions([csv_image.image for csv_image in training_lists], [csv_image.image for csv_image in test_lists])
     print("PCA finished successfully")
 
     # Replace unreduced CsvImage vectors by reduced ones, for training and test images
-    for i in range(len(training_lists[0])):
+    for i in range(len(training_lists)):
         # print(len(training_lists[0][i].image)) # for debugging
-        training_lists[0][i].image = reduced_images[0][i]
+        training_lists[i].image = reduced_images[0][i]
```

Comment:
I would refrain from overwriting here and rather introduce an additional attribute to the class, even though it will double the memory consumtion.

Reply:
This overwriting method was chosen because it allows switching between running the pca or not without the need to change the parameters of later functions, like the knn. Also regarding that this method takes less memory it seems more reasonable to keep doing it this way.